api/src/main/java/com/example/api/service/ranking/RankingService.java

```diff
@@ -0,0 +1,103 @@
+    private final FileTaskResultRepo fileTaskResultRepo;
+    private final GraphTaskResultService graphTaskResultService;
+    private final ProfessorFeedbackService professorFeedbackService;
+
+    public List<RankingEntry> getRanking() {
+        return userRepo.findAllByAccountTypeEquals(AccountType.STUDENT)
+                .stream()
+                .map(this::studentToRankingEntry)
+                .sorted(Comparator.comparingDouble(RankingEntry::getPoints).reversed())
+                .collect(Collectors.toList());
+    }
+
+    public List<RankingEntry> getRankingForGroup(String groupName) {
+        return userRepo.findAllByAccountTypeEquals(AccountType.STUDENT)
+                .stream()
+                .filter(student -> Objects.equals(student.getGroup().getName(), groupName))
+                .map(this::studentToRankingEntry)
+                .sorted(Comparator.comparingDouble(RankingEntry::getPoints).reversed())
+                .collect(Collectors.toList());
+    }
+
+    public List<RankingEntry> getSearchedRanking(String search) {
+        String searchLower = search.toLowerCase();
+        return userRepo.findAllByAccountTypeEquals(AccountType.STUDENT)
+                .stream()
+                .filter(student ->
+                                student.getFirstName().toLowerCase().contains(searchLower) ||
+                                student.getLastName().toLowerCase().contains(searchLower) ||
+                                student.getHeroType().toString().toLowerCase().contains(searchLower) ||
+                                student.getGroup().getName().toLowerCase().contains(searchLower))
+                .map(this::studentToRankingEntry)
+                .sorted(Comparator.comparingDouble(RankingEntry::getPoints).reversed())
+                .collect(Collectors.toList());
+    }
+
+    private RankingEntry studentToRankingEntry(User student) {
+        RankingEntry rankingEntry = new RankingEntry(student);
+        rankingEntry.setPoints(getStudentPoints(student));
+        return rankingEntry;
+    }
+
+    private Double getGraphTaskPoints(User student) {
+        return graphTaskResultRepo.findAllByUser(student)
+                .stream()
+                .mapToDouble(task -> {
+                    try {
+                        return graphTaskResultService.getAndSetAllPoints(task.getId());
+                    } catch (Exception e) {
+                        log.info("GraphTaskResult with id {} not checked yet", task.getId());
+                    }
+                    return 0.0;
+                })
+                .sum();
+    }
+
+    private Double getFileTaskPoints(User student) {
+        return fileTaskResultRepo.findAllByUser(student)
+                .stream()
+                .mapToDouble(task -> {
+                    try {
+                        return professorFeedbackService.getProfessorFeedbackForFileTaskResult(task.getId()).getPoints();
```

Comment:
Tu to samo, jak profesor oceni taska to wszystkie punkty zapisują się w pointsReceived (przynajmniej powinny XD)